askama_shared/src/generator.rs

```diff
@@ -353,11 +353,21 @@ impl<'a> Generator<'a> {
                     self.handle_include(state, ws, path);
                 },
                 Node::Call(ref ws, name, ref args) => self.write_call(state, ws, name, args),
-                Node::Macro(_, _) |
-                Node::Import(_, _) |
+                Node::Macro(_, ref m) => {
+                    if let AstLevel::Nested = level {
+                        panic!("macro blocks only allowed at the top level");
+                    }
+                    self.write_macro(m);
```

Comment:
Can this just be `self.handle_ws(&m.ws)` or some such? (And same for the `Import` case.)